source/org/zfin/task/metricsmonitor/MetricsMonitor.java

```diff
@@ -0,0 +1,192 @@
+
+/**
+ * Uses jmx to connect to tomcat for various tomcat metrics and
+ * Reads /proc/loadavg to get the cpu load averages.
+ * Once the "1 minute load average" exceeds the LOAD_THRESHOLD_HIGH or
+ * the "average number of milliseconds processing time per request" from tomcat exceeds MS_PER_REQUEST_THRESHOLD_HIGH,
+ * it will turn on captcha.
+ *
+ * If the captcha is enabled, it will disable when the relevant metrics drop below the corresponding "_LOW" thresholds
+ * and a certain amount of time has passed (SECONDS_TO_ENABLE)
+ */
+@Log4j2
+public class MetricsMonitor {
+    private static final String TOMCAT_CONNECTOR_NAME = "https-openssl-nio-8443";
+
+    private final SimpleDateFormat timeFormat = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+    private HighLoadAction triggerAction = new HighLoadAction();
+    private JMXConnector jmxConnector;
+    private MBeanServerConnection mbeanConn;
+    private SystemMetricsDTO previousSystemMetrics;
+    private long rowNumber = 0;
+    private MetricsMonitorConfig configuration;
+
+    private void run() throws IOException {
+        this.init();
+        this.connect();
+        this.startMonitoring();
+    }
+
+    private void init() {
+        this.configuration = MetricsMonitorConfig.getInstance();
+    }
+
+    public void connect() throws IOException {
+        log.info("Connecting to " + configuration.jmxUrl());
+        JMXServiceURL url = new JMXServiceURL(configuration.jmxUrl());
+        jmxConnector = JMXConnectorFactory.connect(url);
+        mbeanConn = jmxConnector.getMBeanServerConnection();
+    }
+
+    public void startMonitoring() {
+        while (true) {
+            try {
+                SystemMetricsDTO metrics = collectMetrics();
+                printMetrics(metrics);
+                triggerActionForMetrics(metrics);
+                Thread.sleep(this.configuration.getPollingIntervalSeconds() * 1000);
+            } catch (Exception e) {
+                System.err.println("Error collecting metrics: " + e.getMessage());
+                e.printStackTrace();
+
+                // Try to reconnect
+                try {
+                    System.out.println("Attempting to reconnect...");
+                    disconnect();
+                    connect();
+                } catch (IOException reconnectEx) {
+                    System.err.println("Failed to reconnect: " + reconnectEx.getMessage());
+                }
+            }
+        }
+    }
+
+    private void triggerActionForMetrics(SystemMetricsDTO metrics) {
+        triggerAction.trigger(metrics);
+    }
+
+    private SystemMetricsDTO collectMetrics() {
+        SystemMetricsDTO metricsRow = new SystemMetricsDTO();
+        metricsRow.setCurrentTime(timeFormat.format(new Date()));
+        collectResponseTimeMetrics(metricsRow);
+        collectLoadAverageMetrics(metricsRow);
+        return metricsRow;
+    }
+
+    private void printMetrics(SystemMetricsDTO metrics) {
+        if (rowNumber == 0) {
```

Comment:
Only print this out if log level is debug?
We could configure a specific file for it to write to.

Reply:
We should have a way to turn off this thread just in case something unexpected starts happening